crates/csharp/src/lib.rs

```diff
@@ -1465,7 +1470,7 @@ impl InterfaceGenerator<'_> {
                         private static extern void wasmImportResourceDrop(int p0);
 
                         protected virtual void Dispose(bool disposing) {{
-                            if (Handle != 0) {{
+                            if (disposing && Handle != 0) {{
```

Comment:
Please also remove or comment out ~finalizer code at lines 1479-1481. It doesn't do anything after this change but adds the object into finalizer queue.